elastic/kibana
# [ILM] Data tiers for 7.10

## Summary

Adds support for data tiers in ILM.

## TODO

- [x] validation and callouts for different states (detecting node roles)
- [x] Hook up the new component in all phases
- [x] merge in frozen phase work (https://github.com/elastic/kibana/pull/75968)
- [ ] Get copy review

## Screenshots

<details>
<summary>Screenshots</summary>

**Select expanded**

<img width="1044" alt="Screenshot 2020-09-10 at 17 24 37" src="https://user-images.githubusercontent.com/8155004/92753824-8b84e400-f38a-11ea-846d-c40aaa4786a9.png">


**Default state (warm tier)**

<img width="1053" alt="Screenshot 2020-09-10 at 17 07 49" src="https://user-images.githubusercontent.com/8155004/92752303-1a90fc80-f389-11ea-89f8-1afc3a78340d.png">



**Custom data tier allocation activated**

<img width="1040" alt="Screenshot 2020-09-10 at 17 07 56" src="https://user-images.githubusercontent.com/8155004/92752346-25e42800-f389-11ea-9e3c-a74ae942a002.png">


**Data tier allocation off (migration { enabled: false } } )**

<img width="1035" alt="Screenshot 2020-09-10 at 17 08 03" src="https://user-images.githubusercontent.com/8155004/92752433-398f8e80-f389-11ea-894c-0a3ff8d212bb.png">


**No node roles found warning**

<img width="1083" alt="Screenshot 2020-09-10 at 17 09 48" src="https://user-images.githubusercontent.com/8155004/92752470-401e0600-f389-11ea-9ad9-135bb832d47d.png">


**No node attrs found warning (pre-existing)**

<img width="1025" alt="Screenshot 2020-09-10 at 17 14 08" src="https://user-images.githubusercontent.com/8155004/92752505-4613e700-f389-11ea-9c61-f19e2ba60a60.png">


</details>

## How to test

1. Start Kibana with `yarn es snapshot -E node.attr.rack_id=rack_one` (this will enable you to use the custom allocation selector
2. Navigate to the ILM policy UI and either edit or create a new policy
3. In the create/edit UI make sure to use Warm/Cold/Frozen phase as this is where the new form field was added for selecting data tier allocation
4. Confirm that selecting "Default" does not add any `allocate` action to the phase in question
5. Confirm that selecting "Off" adds `migrate: { enabled: false }` to the policy `actions` section
6. Confirm that selecting "Custom" enables you to select the node attribute you started the cluster with

Note: the "Off" option should only start working once https://github.com/elastic/elasticsearch/pull/61377 is merged.

### Checklist

Delete any items that are not applicable to this PR.


### For maintainers
